packages/dds/tree/src/feature-libraries/sequence-field/tracker.ts

```diff
@@ -84,9 +96,12 @@ export class GapTracker {
 				// Reset the offset for the revisions chronologically after the attach to zero.
 				// This is because for those revisions, the nodes were present in the input context.
 				// In other words, one revision's attach is later revisions' skip.
-				for (const rev of this.map.keys()) {
-					if (rev > revision) {
-						this.map.delete(rev);
+				if (this.map.size > 0) {
+					const revisionIndex = this.revisionIndexer(revision);
+					for (const rev of this.map.keys()) {
+						if (this.revisionIndexer(rev) > revisionIndex) {
+							this.map.delete(rev);
```

Comment:
This wasn't changed in this PR, but is it safe to delete elements from a map while using the `keys` iterator?

Reply:
Good question. Manual testing and stack overflow searches seem to indicate that it is. I can't quite find an authoritative source to confirm it though, and the [standard](https://262.ecma-international.org/13.0/#sec-createmapiterator) doesn't convince me at all that it is. :(

Reply:
Though it's clear from the standard that they do anticipate the collection being edited during iteration.

Reply:
Interestingly, the [standard for forEach](https://262.ecma-international.org/13.0/#sec-map.prototype.foreach) is phrased in a way that gives me more confidence.

Reply:
Ah, I get it now. The standard is using an abstraction where deletion do not reduce the length of the entries list. It just [hollows out the entry being deleted](https://tc39.es/ecma262/#sec-map.prototype.delete). That makes the algorithm I listed in the first link safe.